src/Akka.Streams.Kafka/Stages/Producers/DefaultProducerStage.cs

```diff
@@ -94,15 +99,34 @@ public DefaultProducerStageLogic(IProducerStage<K, V, P, TIn, TOut> stage, Attri
                             {
                                 var result = new TaskCompletionSource<MultiResultPart<K, V>>();
                                 AwaitingConfirmation.IncrementAndGet();
-                                Producer.Produce(record, BuildSendCallback(result, report =>
+                                var callback = BuildSendCallback(
+                                    completion: result,
+                                    onSuccess: report =>
+                                    {
+                                        result.SetResult(new MultiResultPart<K, V>(report, record));
+                                    },
+                                    onFailure: OnProduceFailure);
+                                try
                                 {
-                                    result.SetResult(new MultiResultPart<K, V>(report, record));
-                                }));
-                                return result.Task;
-                            });
-                            PostSend(msg);
-                            var resultTask = Task.WhenAll(tasks).ContinueWith(t => new MultiResult<K, V, P>(t.Result.ToImmutableHashSet(), multiMessage.PassThrough) as IResults<K, V, P>);
-                            Push(stage.Out, resultTask as Task<TOut>);
+                                    Producer.Produce(record, GetAsyncCallback(callback));
+                                    return result.Task;
+                                }
+                                catch (Exception exception)
+                                {
+                                    OnProduceFailure(exception);
```

Comment:
Enclose the Producer.Produce with a try..catch and route any exceptions to the OnProduceFailure method.